gemstone/src/swapper/mayan/relayer.rs

```diff
@@ -0,0 +1,309 @@
+        match self {
+            QuoteType::Swift => write!(f, "SWIFT"),
+            QuoteType::Mctp => write!(f, "MCTP"),
+            QuoteType::Swap => write!(f, "SWAP"),
+            QuoteType::WH => write!(f, "WH"),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+pub struct Quote {
+    #[serde(rename = "type")]
+    pub r#type: String,
+    pub effective_amount_in: f64,
+    pub expected_amount_out: f64,
+    pub price_impact: Option<f64>,
+    pub min_amount_out: f64,
+    pub min_middle_amount: Option<f64>,
+    pub evm_swap_router_address: Option<String>,
+    pub evm_swap_router_calldata: Option<String>,
+    pub min_received: f64,
+    pub gas_drop: f64,
+    pub price: f64,
+    pub swap_relayer_feed: Option<f64>,
+    pub redeem_relayer_fee: Option<f64>,
+    pub refund_relayer_fee: Option<f64>,
+    pub solana_relayer_fee: Option<f64>,
+    pub refund_relayer_fee64: String,
+    pub cancel_relayer_fee64: String,
+    pub from_token: Token,
+    pub to_token: Token,
+    pub from_chain: String,
+    pub to_chain: String,
+    pub slippage_bps: u32,
+    pub bridge_fee: f64,
+    pub suggested_priority_fee: f64,
+    pub only_bridging: bool,
+    pub deadline64: String,
+    pub referrer_bps: Option<u32>,
+    pub protocol_bps: Option<u32>,
+    pub swift_mayan_contract: Option<String>,
+    pub swift_auction_mode: Option<u8>,
+    pub swift_input_contract: String,
+    pub swift_input_decimals: u8,
+    pub gasless: bool,
+    pub relayer: String,
+    pub send_transaction_cost: f64,
+    pub max_user_gas_drop: f64,
+}
+
+#[derive(Debug, Deserialize)]
+struct QuoteResponse {
+    quotes: Vec<Quote>,
+
+    #[serde(rename = "minimumSdkVersion")]
+    pub minimum_sdk_version: Vec<u8>,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum MayanRelayerError {
+    #[error("Network error: {0}")]
+    NetworkError(String),
+    #[error("Invalid response: {0}")]
+    InvalidResponse(String),
+    #[error("SDK version not supported")]
+    SdkVersionNotSupported,
+    #[error("Invalid parameters: {0}")]
+    InvalidParameters(String),
+}
+
+#[derive(Debug)]
+pub struct MayanRelayer {
+    url: String,
+    provider: Arc<dyn AlienProvider>,
+}
+
+impl MayanRelayer {
+    pub fn new(url: String, provider: Arc<dyn AlienProvider>) -> Self {
+        Self { url, provider }
+    }
+
+    pub fn default_relayer(provider: Arc<dyn AlienProvider>) -> Self {
+        Self::new("https://price-api.mayan.finance/v3".to_string(), provider)
+    }
+
+    pub async fn get_quote(&self, params: QuoteParams, options: Option<QuoteOptions>) -> Result<Vec<Quote>, MayanRelayerError> {
+        let options = options.unwrap_or_default();
+        let from_chain = if params.from_chain == Chain::SmartChain {
+            "bsc".to_string()
+        } else {
+            params.from_chain.to_string()
+        };
+
+        let mut query_params = vec![
```

Comment:
small improvement: make this params as a `struct`, `serde_urlencoded` supports too